__tests__/UITests.tsx

```diff
@@ -104,15 +104,64 @@ describe("LibraryContainer UI", function () {
     chai.expect(libraryItem.state('expanded')).to.be.true; // check for if the libraryItem is expanded after mouse click       
   });
 
+  it("should raise the onItemWillExpand when expanding", function (done) {
+    // test data to create LibraryItem  
+    // parent item that will hold the child items
+    let data = new ItemData("");
+    data.text = "TestItem";
+    data.itemType = "none";
+
+    // create child items and link with parent
+    for (var i = 0; i < 2; i++) {
+      let item = new ItemData("");
+      item.text = "Child" + i;
+      item.itemType = "category";
+      data.appendChild(item);
+    }
+
+    // create "LibraryContainer" to pass as an argument for creation of "LibraryItem"
+    let libContainer = LibraryEntryPoint.CreateLibraryController();
+    //pass a callback which if called will end the test
+    let libraryItem = mount(<LibraryItem libraryContainer={libContainer} data={data} showItemSummary={false} onItemWillExpand={() => { done() }} />);
+
+    let header = libraryItem.find(('div.LibraryItemHeader')).at(0);// the state of LibraryItem is changed when clicking on header
+    chai.expect(header).to.have.lengthOf(1); // verify that there is a header
+    header.simulate('click'); // enzyme function to simulate mouse click
+    chai.expect(libraryItem.state('expanded')).to.be.true;
+  });
+
+  it("scrollToElement should be called when libraryItem is expanded only", function () {
+
+    // Test for positive scenario where the node names are correct
+    let libContainer = mount(
+      libController.createLibraryContainer()
+    );
+    // Load the data to libController
+    libController.setLoadedTypesJson(loadedTypesJson, false);
+    libController.setLayoutSpecsJson(layoutSpecsJson, false);
+    libController.refreshLibraryView();
+
+    let header = libContainer.find('div.LibraryItemHeader').at(0);
+    chai.expect(header).to.have.lengthOf(1);
+    var count = 0;
+    //replace the scroll method with a method which ends the test.
+    libContainer.getNode().scrollToExpandedItem = () => { count = count + 1; }
```

Comment:
Is it because we do not have sinon or similar dependency which can mock this for us?

Reply:
I could attempt it that way, I think sinon or something with a spy API is loaded, I just prefer this for simplicity and a new entity is built for each test so it should not effect other tests. Can try to use a spy if you prefer.

Reply:
@QilongTang I lied, I don't see any use of spies in the repo with a quick search, must have been somewhere else.

Reply:
OK. I don't mean that we need to introduce that as part of this PR, was just curious